### PR TITLE
cancel previous auto-close when opened changes

### LINKED ISSUE #48: debounced close should be canceled
each time `opened` change, the auto-close debounce should be canceled, and eventually reset.

### PR DESCRIPTION
**Pull request by valdrinkoshi**
Fixes #48 by canceling previous auto-closing of the toast

### CODE REVIEW

File: paper-toast.html [hunk 1]
```diff
@@ -187,6 +187,7 @@
          * Called when the value of `opened` changes.
          */
         _openedChanged: function() {
+          this.cancelDebouncer('auto-close');
```
Comment (cdata) [inline paper-toast.html]: If we always cancel the debouncer, does it need to be `debounce`? Or should it be `async`?

Comment (valdrinkoshi) [inline paper-toast.html]: You're so right..! Will update!

**Comment by valdrinkoshi** [comment]
@cdata, feedback incorporated :beers:

File: paper-toast.html [hunk 2]
```diff
@@ -187,6 +187,8 @@
          * Called when the value of `opened` changes.
          */
         _openedChanged: function() {
+          this.cancelAsync(this._autoClose);
+          this._autoClose = undefined;
```
Comment (cdata) [inline paper-toast.html]: Conventionally we use `null` for a member property that has no value.

File: test/basic.html [hunk 3]
```diff
@@ -9,106 +9,156 @@
...
+        toast2.open();
+        toast1.open();
+        assert.isTrue(toast1.opened, 'toast1 is opened');
+        assert.isFalse(toast2.opened, 'toast2 is not opened');
+        toast2.open();
+        assert.isFalse(toast1.opened, 'toast1 is now not opened');
+        assert.isTrue(toast2.opened, 'toast2 is now opened');
+      });
+
+      test('toast gets closed only once when doing show/hide', function(done) {
```
Comment (cdata) [inline test/basic.html]: Is it valuable to test multiple calls to `hide`?

**Comment by cdata** [comment]
Thanks for the work on this. The tests are looking particularly suite :hear_no_evil: 

Some more minor feedback - PTAL!

File: paper-toast.html (hunk 2 above)
Comment (valdrinkoshi) [inline paper-toast.html]: is there any advantage in setting it to `null` (e.g. performance)?
I was tempted to do
```javascript
this._autoClose = undefined;
delete this._autoClose;

File: test/basic.html (hunk 3 above)
Comment (valdrinkoshi) [inline test/basic.html]: this test is valuable with the next one - probably the should be in the same suite.
The final goal is to keep the toast opened if the last action performed was `show()` or `open()`. I've separated the test in 2 checks:
- check that close is invoked once
- check that the toast is opened after a series of open/close/open

WDYT? too many checks?

File: paper-toast.html (hunk 2 above)
Comment (valdrinkoshi) [inline paper-toast.html]: also, `cancelAsync` doesn't like `null` as input, so it would require an extra-check :x

**Comment by valdrinkoshi** [comment]
@cdata I made the test more "readable" & let some comments in here, PTAL

**Comment by cdata** [comment]
LGTM :+1: nice work!

**Pull request merged**